### DISCUSSION TITLE
openai whisper model Integration with fetch.ai micro-Agents

### DISCUSSION THREAD

**Discussion by abhi141** [fetchai/uAgents#154] [Integrations]
Hello Enthusiasts,

I've noticed the varied integrations in your repo and came across [openai whisper-large-v2](https://huggingface.co/openai/whisper-large-v2). I'm keen on adding the Whisper integration to the mix.

About the model - Whisper is a pre-trained model for automatic speech recognition (ASR) and speech translation. Trained on 680k hours of labelled data, Whisper models demonstrate a strong ability to generalise to many datasets and domains without the need for fine-tuning.

Looking forward to the possibility of contributing!

**Comment by jrriehl** [comment]
That sounds great @abhi141! Please feel free to create a PR with this integration.

**Reply by abhi141** [reply]
Thanks @jrriehl, I have raised a PR. Please have a look and let me know if I can improve it in any way.

**Comment by rishankjhavar** [comment]
Wow, this sounds promising. I can think of a lot of uagent-based apps that could utilize this. Please feel free to raise a PR @abhi141!

**Comment by xsa-dev** [comment]
I started making the same topic, but I didn’t know that you had already started development, I posted what is now ready:
- implementation with Hugging Face
- https://github.com/xsa-dev/uAgents/tree/whisper-something-unfinished/integrations/whisper
- not tested
I would be glad to do something, but I don’t know how to take the task, if I can add it, I can help

**Comment by abhifetch** [comment]
Hi @xsa-dev ,

this integration look good. we will take reference from your gitHub repo and make similar integration included in our uAgents repo as per standards
